### PR TITLE
Feature animation

### PR DESCRIPTION
**Pull request by KRademacher**

### CODE REVIEW

File: Assets/_Scenes/Scene_Combat/Scripts/Combat/MinionSpawner.cs [hunk 1]
```diff
@@ -11,16 +12,29 @@ public class MinionSpawner : MonoBehaviour
...
     private Minion spawnedMinion;
     private GameObject spawnedMinionPrefab;
     private NavMeshAgent spawnedMinionAgent;
+    private Animator animator;
 
     private void Start()
     {
         boss = GameObject.FindGameObjectWithTag("Boss");
         nodeController = GameObject.Find("NodeController").GetComponent<NodeController>();
+        particleSystems = summoningCircle.GetComponentsInChildren<ParticleSystem>().OrderBy(p => p.main.duration).ToArray();
```
Comment (mrtsven) [inline Assets/_Scenes/Scene_Combat/Scripts/Combat/MinionSpawner.cs]: A bit of whitespace is welcome

File: Assets/_Scenes/Scene_Combat/Scripts/Combat/MinionSpawner.cs [hunk 2]
```diff
@@ -66,12 +80,12 @@ private void Attack()
...
     private bool AgentReachedDestination()
     {
-        if (!spawnedMinionPrefab.activeInHierarchy)
+        if (!spawnedMinionAgent || !spawnedMinionPrefab.activeInHierarchy)
         {
             return false;
         }
         //Courtesy of https://answers.unity.com/questions/324589/how-can-i-tell-when-a-navmesh-has-reached-its-dest.html
-        if (!spawnedMinionAgent.pathPending) //Is agent still on a path?
+        if (spawnedMinionAgent.hasPath) //Is agent still on a path?
```
Comment (mrtsven) [inline Assets/_Scenes/Scene_Combat/Scripts/Combat/MinionSpawner.cs]: Don't forget to change the comment as hasPath and pathPending is different.

File: Assets/_Scenes/Scene_Combat/Scripts/Combat/ScriptableObjects/MinionStrong.asset [hunk 3]
```diff
@@ -16,5 +16,5 @@ MonoBehaviour:
   Model: {fileID: 5083813676927505262, guid: 645716cdbd915db48bb3e5abeb0b0869, type: 3}
   Type: 0
   Damage: 3
-  HasSpawned: 0
+  HasSpawned: 1
```
Comment (mrtsven) [inline Assets/_Scenes/Scene_Combat/Scripts/Combat/ScriptableObjects/MinionStrong.asset]: Hmm, no?

Comment (KRademacher) [reply Assets/_Scenes/Scene_Combat/Scripts/Combat/ScriptableObjects/MinionStrong.asset]: Fixed

File: Assets/_Scenes/Scene_Combat/Scripts/Combat/MinionSpawner.cs (hunk 2 above)
Comment (KRademacher) [reply Assets/_Scenes/Scene_Combat/Scripts/Combat/MinionSpawner.cs]: Done.

File: Assets/_Scenes/Scene_Combat/Scripts/Combat/MinionSpawner.cs (hunk 1 above)
Comment (KRademacher) [reply Assets/_Scenes/Scene_Combat/Scripts/Combat/MinionSpawner.cs]: Added whitespaces in new commit.

**Pull request merged**